desktop-widgets/diveplanner.cpp

```diff
@@ -232,10 +244,42 @@ void DivePlannerWidget::heightChanged(const int height)
 	plannerModel->setSurfacePressure(pressure);
 }
 
-void DivePlannerWidget::salinityChanged(const double salinity)
+void DivePlannerWidget::waterTypeUpdateTexts()
 {
-	/* Salinity is expressed in weight in grams per 10l */
-	plannerModel->setSalinity(lrint(10000 * salinity));
+	double density;
+	for (int i = 0; i < 3; i++) {
+		if (ui.waterType->itemData(i) != QVariant::Invalid) {
+			QString densityText = ui.waterType->itemText(i).split("(")[0].trimmed();
+			density = ui.waterType->itemData(i).toDouble();
+			densityText.append(QString(" (%1kg/l)").arg(QString::number(density, 'f', 2)));
+			ui.waterType->setItemText(i, densityText);
+		}
+	}
+}
+
+void DivePlannerWidget::waterTypeChanged(const int index)
+{
+	bool ok = true;
+	double density;
+	
+	if (index == 2) {
+#if QT_VERSION >= QT_VERSION_CHECK(5, 10, 0)
+		/* Defining step width with input mode getDouble is only supported in QT versions >= 5.10 */
+		density = QInputDialog::getDouble(this, tr("Custom water type"),
+			tr("Water density in kg/l:"), 1.00, 0.99, 1.30, 2, &ok,  QFlags<Qt::WindowType>(), 0.01);
```

Comment:
Could we remember the custom density and use that as starting value?

Reply:
I like the idea! Do you think it should be stored in the dive log or something like a local cookie?